### PR TITLE
fix: correct Google GenAI SDK imports and API usage

### PR DESCRIPTION
**Pull request by J-MaFf**
## Summary

Complete fix for Google GenAI SDK integration and test suite. Resolves all import errors and brings application to full production readiness with 100% test coverage.

### Core SDK Fixes
- Fixed google.generativeai SDK imports to use proper submodule structure:
  - `from google.generativeai.client import configure`
  - `from google.generativeai.generative_models import GenerativeModel`
  - `from google.generativeai import types`
- Added `# type: ignore` comments for type checker compatibility
- Updated all API calls to use new SDK pattern: `configure()` → `GenerativeModel()` → `model.generate_content()`
- Changed Gemini model from `gemini-2.5-flash-001` to `gemini-flash-lite-latest` (always latest, widely available)

### Configuration Enhancements
- Added `team_id` field to `ClickUpConfig` with default value `'9014534294'`
- Implemented Team ID fallback logic in `extractor.py` for `/team` endpoint failures
- Team ID can be overridden via `CLICKUP_TEAM_ID` environment variable
- Eliminates need for user prompts when Team ID is available

### Bug Fixes
- Fixed UnboundLocalError in `extractor.py` where `team` variable wasn't initialized before use
- Added fallback team object creation when using Team ID directly
- Removed debug logging that exposed API key prefixes

### Test Suite Updates (165/165 tests passing ✅)
- **test_ai_summary_success.py**: Updated 19 tests from old `genai.Client()` mocks to new SDK pattern
- **test_extractor.py**: Added `team_id` parameter to config in test fixtures
- **test_extractor_edge_flows.py**: Fixed workspace error handling test with proper mock responses
- All tests verified passing with new Google GenAI SDK implementation

### Production Verification
- Application tested end-to-end: successfully exports 22 tasks with all features enabled
- AI summary generation working with Gemini model
- ClickUp API v2 authentication and task extraction verified
- 1Password SDK integration confirmed functional
- All output formats (HTML, CSV, Markdown, PDF) working

## Technical Details

**SDK Version**: google-generativeai 0.8.5+

**Breaking Changes from Old API**:
- Old: `genai.Client(api_key=key).models.generate_content()`
- New: `configure(api_key=key); GenerativeModel(name).generate_content(prompt, generation_config=config)`

**Why This Fix**:
The installed google-generativeai package v0.8.5 uses a different API structure than the old `genai` library. The configure/GenerativeModel pattern is the correct way to use this SDK version.

## Testing

- ✅ 165 unit tests passing
- ✅ End-to-end application test: 22 tasks exported successfully
- ✅ AI summary generation verified working
- ✅ ClickUp API authentication working
- ✅ Configuration fallback logic validated
- ✅ Error handling and edge cases covered

## Migration Notes

No user-facing changes required. Application uses the same CLI interface:
```bash
python main.py --workspace "KMS" --space "Kikkoman" --ai-summary
```

Team ID can be set via environment variable for non-interactive mode:
```bash
$env:CLICKUP_TEAM_ID = "9014534294"
python main.py
```

### CODE REVIEW

File: tests/test_ai_summary_success.py [hunk 1]
```diff
@@ -63,107 +63,112 @@ def test_no_api_key_returns_field_block(self):
...
+
         result = get_ai_summary('Test Task', field_entries, '')
-        
+
         self.assertIn('Subject: Test subject', result)
         self.assertIn('Description: Test description', result)
         self.assertIn('Resolution: (not provided)', result)
 
-    @patch('ai_summary.genai', None)
+    @patch('ai_summary.configure', side_effect=Exception('SDK not available'))
```
[the remark this answers is not in the document]
Comment (J-MaFf) [reply tests/test_ai_summary_success.py]: @copilot open a new pull request to apply changes based on [this feedback](https://github.com/J-MaFf/clickup_task_extractor/pull/46#discussion_r2539403726)
